### ISSUE TITLE
Clippy links to `master` instead of actual version

### ISSUE THREAD

**Issue by Lakelezz** [rust-lang/rust-clippy#4167]
Hello everyone,

whenever Clippy suggests a change, it links to further information on its website.
However the link always directs to master, hence if a lint-name changes, the link becomes invalid.

One example is `cognitive_complexity`.

We can find it on `master`: https://rust-lang.github.io/rust-clippy/master/index.html#cognitive_complexity
But not on `v0.0.212`, as there seems to be a name-change: https://rust-lang.github.io/rust-clippy/v0.0.212/index.html#cyclomatic_complexity

It once linked me: https://rust-lang.github.io/rust-clippy/master/index.html#cyclomatic_complexity
To no success obviously.

It would be great if Clippy would link to the correct version. E.g. `v0.0.212` links to https://rust-lang.github.io/rust-clippy/v0.0.212
I would be willingly to contribute this change too!

Thanks for your time, keep the great work up : )

**Comment by mati865** [comment]
The `0.0.212` is one year old and there is no plan to release Clippy versions because it's tied to Rust internals too much to make any sense.

IMO better solution would be to make old links redirect to the new one when lint is renamed.

**Comment by Manishearth** [comment]
We need to completely overhaul the versioning here.

For now I think we should get rid of the old versions from the list and make the homepage redirect to `/master`, and then get it to deal with renames. After that, we can see if we can hook this up to rust's release system so that we get proper docs. Though I'm also fine with only maintaining docs for master.

**Comment by Lakelezz** [comment]
> The `0.0.212` is one year old and there is no plan to release Clippy versions because it's tied to Rust internals too much to make any sense.

Oh! I just checked on Github and yes, it really is a year old. 
I was simply doing: `cargo clippy --version` and it gave me: `clippy 0.0.212 (265318db 2019-05-17)`
That caused me to think it was not too old. 

Having no versions might make it hard to refer to a specific version then. If I use a certain toolchain version, am I guaranteed that everyone is using the same Clippy?

**Comment by Manishearth** [comment]
Yeah so the clippy version is actually the _Rust_ version (try `rustc -V`).. As long as everyone has the same `rustc` you will have the same Clippy.
